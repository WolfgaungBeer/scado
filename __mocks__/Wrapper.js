import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { Theme, defaultTheme } from '../scado';

const mockStore = configureStore();
const store = mockStore({
    scado: {
        theme: defaultTheme,
        toasts: [
            { id: '1', type: 'success', content: 'Text', onClose: () => {} },
            { id: '2', type: 'error', content: 'Text', onClose: () => {} },
        ],
    },
});


const Wrapper = ({ children }) => ( // eslint-disable-line
    <Provider store={store}>
        <Theme>{children}</Theme>
    </Provider>
);

export default Wrapper;
