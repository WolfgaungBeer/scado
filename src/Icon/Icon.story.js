import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { text, boolean } from '@storybook/addon-knobs';
import Icon from './';


storiesOf('Icon', module)
    .add('default', () => (
        <Icon icon="check_circle" />
    ))
    .add('with props', () => (
        <Icon
            icon={text('icon', 'check_circle')}
            color={text('color', 'success')}
            scale={text('scale', 'l')}
            loading={boolean('loading', false)}
            onClick={action('icon clicked')}
        />
    ));
