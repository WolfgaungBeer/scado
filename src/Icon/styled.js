import styled, { keyframes, css } from 'styled-components';

const rotationKeyframes = keyframes`
    from { transform: rotate(0deg); }
    to { transform: rotate(360deg); }
`;

export const StyledIcon = styled.i`
    cursor: ${props => props.onClick && 'pointer'};
    vertical-align: -16%;
    padding: 0 0.25rem;
    color: ${props => props.theme.color[props.color] || 'inherit'};
    font-size: ${props => props.theme.scale[props.scale] || 'inherit'};
    ${(props) => {
        if (props.loading) {
            return css`
                display: inline-block;
                animation: ${rotationKeyframes} 0.85s linear infinite;
            `;
        }
        return undefined;
    }};
`;
