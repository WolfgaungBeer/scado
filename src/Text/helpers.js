export const textStyle = () => props => `
    color: ${props.theme.color[props.color]};
    font-size: ${props.theme.scale[props.scale]};
`;

export const font = type => props => `
    font-family: ${props.theme.fonts[type]};
`;
