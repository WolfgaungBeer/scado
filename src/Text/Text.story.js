import React from 'react';
import { storiesOf } from '@storybook/react';
import { text } from '@storybook/addon-knobs';
import Text from './';
import Icon from '../Icon';


storiesOf('Text', module)
    .add('H1', () => (
        <Text.H1>Das ist ein Text</Text.H1>
    ))
    .add('H2', () => (
        <Text.H2>Das ist ein Text</Text.H2>
    ))
    .add('H3', () => (
        <Text.H3>Das ist ein Text</Text.H3>
    ))
    .add('H4', () => (
        <Text.H4>Das ist ein Text</Text.H4>
    ))
    .add('H5', () => (
        <Text.H5>Das ist ein Text</Text.H5>
    ))
    .add('H6', () => (
        <Text.H6>Das ist ein Text</Text.H6>
    ))
    .add('P', () => (
        <Text.P>Das ist ein Text</Text.P>
    ))
    .add('A', () => (
        <Text.A>Das ist ein Text</Text.A>
    ))
    .add('Label', () => (
        <Text.Label>Das ist ein Text</Text.Label>
    ))
    .add('with props', () => (
        <Text.H1
            color={text('color', 'success')}
            scale={text('scale', 'l')}
        >Das ist ein Text</Text.H1>
    ))
    .add('with an icon on the left side', () => (
        <div>
            <Icon icon="dashboard" color="success" scale="xl" />
            <Text.Span color="success" scale="xl">Das ist ein Text</Text.Span>
        </div>
    ))
    .add('with an icon on the right side', () => (
        <div>
            <Text.Span color="success" scale="xl">Das ist ein Text</Text.Span>
            <Icon icon="dashboard" color="success" scale="xl" />
        </div>
    ))
    .add('with an nested icon', () => (
        <Text.Span color="success" scale="xl">
            <Icon icon="dashboard" color="success" scale="xl" />
            Das ist ein Text
        </Text.Span>
    ));
