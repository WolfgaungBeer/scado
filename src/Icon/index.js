import React from 'react';
import { string, bool, func, node } from 'prop-types';
import { StyledIcon } from './styled';

const propTypes = {
    color: string,
    scale: string,
    className: string,
    loading: bool,
    onClick: func,
    children: node,
};

const defaultProps = {
    color: undefined,
    scale: undefined,
    className: undefined,
    loading: undefined,
    onClick: undefined,
    children: undefined,
};

const Icon = ({ color, scale, className, loading, onClick, children }) => {
    const classname = `material-icons ${className || ''}`;
    const iconValue = loading ? 'autorenew' : children;
    return (
        <StyledIcon className={classname} color={color} scale={scale} loading={loading} onClick={onClick}>
            {iconValue}
        </StyledIcon>
    );
};

Icon.propTypes = propTypes;
Icon.defaultProps = defaultProps;

export default Icon;
