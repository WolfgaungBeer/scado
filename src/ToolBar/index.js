import React from 'react';
import { string, node } from 'prop-types';
import { StyledToolBar, ChildrenWrapper } from './styled';


const propTypes = {
    className: string,
    children: node,
    actionButtons: node,
};

const defaultProps = {
    className: undefined,
    children: undefined,
    actionButtons: undefined,
};

const ToolBar = ({ className, children, actionButtons }) => (
    <StyledToolBar alignItems="center" justifyContent="space-between" className={className}>
        <ChildrenWrapper alignItems="center">{children}</ChildrenWrapper>
        {actionButtons && <ChildrenWrapper alignItems="center">{actionButtons}</ChildrenWrapper>}
    </StyledToolBar>
);

ToolBar.propTypes = propTypes;
ToolBar.defaultProps = defaultProps;

export default ToolBar;
