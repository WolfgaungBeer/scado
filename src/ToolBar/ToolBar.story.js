import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import ToolBar from './';
import Icon from '../Icon';
import Text from '../Text';

const ActionButtons = () => (
    <div>
        <Icon color="white" scale="l" onClick={action('search clicked')}>search</Icon>
        <Icon color="white" scale="l" onClick={action('settings clicked')}>settings</Icon>
        <Icon color="white" scale="l" onClick={action('more button clicked')}>keyboard_arrow_down</Icon>
    </div>
);

storiesOf('ToolBar', module)
    .add('default', () => (
        <ToolBar actionButtons={<ActionButtons />}>
            <Icon color="white" scale="xl" onClick={action('menu clicked')}>dashboard</Icon>
            <Text.Span color="white" scale="l" onClick={action('text clicked')}>Massiv App</Text.Span>
        </ToolBar>
    ));
