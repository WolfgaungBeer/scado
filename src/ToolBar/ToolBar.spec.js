import React from 'react';
import renderer from 'react-test-renderer';
import Wrapper from '../../__mocks__/Wrapper';
import ToolBar from './';
import Icon from '../Icon';
import Text from '../Text';

const click = () => {};

const ActionButtons = () => (
    <div>
        <Icon color="white" scale="l" onClick={click}>search</Icon>
        <Icon color="white" scale="l" onClick={click}>settings</Icon>
        <Icon color="white" scale="l" onClick={click}>keyboard_arrow_down</Icon>
    </div>
);

it('renders the ToolBar Component correctly', () => {
    const tree = renderer.create(
        <Wrapper>
            <ToolBar actionButtons={<ActionButtons />}>
                <Icon color="white" scale="xl" onClick={click}>dashboard</Icon>
                <Text.Span color="white" scale="l" onClick={click}>Massiv App</Text.Span>
            </ToolBar>
        </Wrapper>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
