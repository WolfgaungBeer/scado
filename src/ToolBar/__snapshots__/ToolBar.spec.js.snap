// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the ToolBar Component correctly 1`] = `
<div
  className="sc-ckVGcZ dCfYYM"
>
  <div
    className="sc-jKJlTe hPcRhJ"
  >
    <i
      className="material-icons  sc-ifAKCX dWDutD"
      color="white"
      onClick={[Function]}
      scale="xl"
    >
      dashboard
    </i>
    <span
      className="sc-jzJRlG aYdSs"
      color="white"
      onClick={[Function]}
      scale="l"
    >
      Massiv App
    </span>
  </div>
  <div
    className="sc-jKJlTe hPcRhJ"
  >
    <div>
      <i
        className="material-icons  sc-ifAKCX fLZEpL"
        color="white"
        onClick={[Function]}
        scale="l"
      >
        search
      </i>
      <i
        className="material-icons  sc-ifAKCX fLZEpL"
        color="white"
        onClick={[Function]}
        scale="l"
      >
        settings
      </i>
      <i
        className="material-icons  sc-ifAKCX fLZEpL"
        color="white"
        onClick={[Function]}
        scale="l"
      >
        keyboard_arrow_down
      </i>
    </div>
  </div>
</div>
`;
